skeletonkey/config.py

```diff
@@ -282,34 +369,18 @@ def add_args_from_dict(
                 )
 
 
-def dict_to_namespace(dictionary: dict) -> argparse.Namespace:
-    """
-    Convert a dictionary to an argparse.Namespace object recursively.
-
-    Args:
-        dictionary (dict): The dictionary to be converted.
-
-    Returns:
-        argparse.Namespace: A Namespace object representing the input dictionary.
-    """
-    for key, value in dictionary.items():
-        if isinstance(value, dict):
-            dictionary[key] = dict_to_namespace(value)
-    return argparse.Namespace(**dictionary)
-
-
-def namespace_to_nested_namespace(namespace: argparse.Namespace) -> argparse.Namespace:
+def key_to_nested_key(key: Key) -> Key:
     """
-    Convert an argparse.Namespace object with 'key1.keyn' formatted keys into a nested Namespace object.
+    Convert an Key object with 'key1.keyn' formatted keys into a nested Key object.
 
     Args:
-        namespace (argparse.Namespace): The Namespace object to be converted.
+        key (Key): The Key object to be converted.
 
     Returns:
-        argparse.Namespace: A nested Namespace representation of the input Namespace object.
+        Key: A nested Key representation of the input Key object.
     """
     nested_dict = {}
-    for key, value in vars(namespace).items():
+    for key, value in vars(key).items():

```

Comment:
This feels weird. I have feeling that this might cause issues. Maybe we should use a different name other that "key".

Reply:
Maybe "ConfigVault" or something like that would be better. (And metaphorically consistent).